File: examples/javascript/linked_list_1.js
class Node {
  constructor(val) { this.val = val; this.next = null; }
}

class LinkedList {
  constructor() { this.head = null; }
  append(val) {
    const node = new Node(val);
    if (!this.head) { this.head = node; return; }
    let curr = this.head;
    while (curr.next) curr = curr.next;
    curr.next = node;
  }
  print() {
    const vals = [];
    let curr = this.head;
    while (curr) { vals.push(curr.val); curr = curr.next; }
    console.log(vals.join(" -> "));
  }
}

const list = new LinkedList();
list.append(1); list.append(2); list.append(3);
list.print();

File: examples/csharp/LinkedList1.cs
using System;

class LinkedList1
{
    class Node { public int Data; public Node Next; }

    static void Main()
    {
        Node head = null;
        head = Insert(head, 3);
        head = Insert(head, 1);
        head = Insert(head, 2);
        Print(head);
    }

    static Node Insert(Node head, int data)
    {
        Node node = new Node { Data = data, Next = head };
        return node;
    }

    static void Print(Node head)
    {
        while (head != null) { Console.Write(head.Data + " "); head = head.Next; }
        Console.WriteLine();
    }
}
